### ISSUE TITLE
[Feature Request] Take a long screenshot when scrolling screen

### ISSUE THREAD

**Issue by linsui** [cvzi/ScreenshotTile#19]
This is a convenient feature on some ROM but I'm not sure if this is possible for an app. Thanks anyway.

**Comment by cvzi** [comment]
I have been working on this for some time. It's possible but a little difficult.
I am not sure if I will get it to work smooth enough to release it 🙄

**Comment by lipici** [comment]
Bump

**Comment by 57382** [comment]
Bump2

**Comment by OkyDooky** [comment]
I want to third this suggestion. My other phone is a newer one with a stock ROM (unrooted) that has a built in "auto-scroll" mode for its screenshot function. It's been really useful (except, it doesn't work with all apps).
@cvzi When you say it's "possible," what methods are you thinking of?

**Comment by cvzi** [comment]
My phone has the same auto-scroll mode and it also doesn't work in all apps.

I tried to replicate that mode.

I have tried to use a [accessibility service](https://developer.android.com/guide/topics/ui/accessibility/service#using-actions) to scroll and then take a screenshot after scrolling.
It didn't work because scrolling "one page" was not possible at all: in one app it would scroll a few "lines" and in other apps the same command would scroll several "pages". 

I guess I could try to add it with manual scrolling. You scroll with your finger and then tap a button to take a screenshot, then scroll with your finger, then tap a button to take a screenshot and so forth. The app could then combine all the images into one image.

**Comment by OkyDooky** [comment]
> My phone has the same auto-scroll mode and it also doesn't work in all apps.
> 
> I tried to replicate that mode.
> 
> I have tried to use a [accessibility service](https://developer.android.com/guide/topics/ui/accessibility/service#using-actions) to scroll and then take a screenshot after scrolling. It didn't work because scrolling "one page" was not possible at all: in one app it would scroll a few "lines" and in other apps the same command would scroll several "pages".

Yeah, the built-in feature has a similar issue, except you can preview its progress and manually stop it short. But, your other option...
> I guess I could try to add it with manual scrolling. You scroll with your finger and then tap a button to take a screenshot, then scroll with your finger, then tap a button to take a screenshot and so forth. The app could then combine all the images into one image.

...is exactly what I was going to suggest. Sort of like the panorama feature in most camera apps. That might be the easiest route to go. And it may be the best, since you have more room to correct and improve things in the future.

**Comment by Anonymous2716** [comment]
This been already implemented or will see in the future? @cvzi

**Comment by magnus-ISU** [comment]
Not adding anything, but until implemented, Stitch on Fdroid can stitch together multiple screenshots into one

**Comment by OkyDooky** [comment]
Maybe whatever [Stitch](https://github.com/ekibun/Stitch)  is using (OpenCV is what the F-Droid page says) could be used as the base for the feature we want here. At least, it could be used for the stitching part, then the auto-repeated screenshots woukd be added in addition to that.

**Comment by rafnov** [comment]
My system screenshot app on Realme devices fails on scrollable screenshot. 
My testing apps for that is anything related with ChatGPT replacements. 
I dont want to use heavy and tracking original app, and affair some screenshots failed there too (probably: "bottom of the page already reached" message.
I've tested Screenshot Tile with Perplexity and ChatGPT in browsers and with gptAssist too. 
Screenshot tile failed with message that some content may have negative impact on scrolling screenshot (native mode) same as system app, or didn't want to scroll at all (old method).
Would be amazing thing to take long shots or select the whole page area, horizontally too, as ChatGPT (reminding that is mostly used testing area, as some other apps refused or have broken screenshots too) something summarises info in wide table.
